sdk/core/Azure.Core/tests/ModelReaderWriter/ModelReaderWriterRequestContentTests.cs

```diff
@@ -0,0 +1,67 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Net.ClientModel;
+using System.Net.ClientModel.Core;
+using System.Reflection;
+using Azure.Core.Tests.ModelReaderWriterTests.Models;
+using NUnit.Framework;
+
+namespace Azure.Core.Tests.ModelReaderWriterTests
+{
+    public class ModelReaderWriterRequestContentTests
+    {
+        private const string json = "{\"kind\":\"X\",\"name\":\"Name\",\"xProperty\":100}";
+        private ModelX _modelX;
+
+        [OneTimeSetUp]
+        public void OneTimeSetUp()
+        {
+            _modelX = ModelReaderWriter.Read<ModelX>(BinaryData.FromString(json));
+        }
+
+        [Test]
+        public void CanCalculateLength()
+        {
+            //use IModelSerializable
+            var content = RequestContent.Create(PipelineContent.CreateContent((IModel<ModelX>)_modelX));
+            AssertContentType(content, "ModelWriterContent");
+            content.TryComputeLength(out long lengthNonJson);
+            Assert.Greater(lengthNonJson, 0);
+
+            //use IModelJsonSerializable
+            var jsonContent = RequestContent.Create(PipelineContent.CreateContent((IJsonModel<ModelX>)_modelX));
+            AssertContentType(jsonContent, "JsonModelWriterContent");
+            content.TryComputeLength(out long lengthJson);
+            Assert.Greater(lengthJson, 0);
+
+            Assert.AreEqual(lengthNonJson, lengthJson);
+
+            //use default
+            jsonContent = RequestContent.Create(_modelX);
+            AssertContentType(jsonContent, "JsonModelWriterContent");
+            content.TryComputeLength(out lengthJson);
+            Assert.Greater(lengthJson, 0);
+
+            Assert.AreEqual(lengthNonJson, lengthJson);
+        }
+
+        private static void AssertContentType(RequestContent content, string expectedContent)
+        {
+            Assert.AreEqual("PipelineContentContent", content.GetType().Name);
+            var field = content.GetType().GetField("_content", BindingFlags.NonPublic | BindingFlags.Instance);
+            Assert.IsNotNull(field);
+            Assert.AreEqual(expectedContent, field.GetValue(content).GetType().Name);
+        }
+
+        [Test]
+        public void ValidatePrivateClassType()
+        {
+            IModel<ModelX> modelX = _modelX;
+
+            RequestContent content = RequestContent.Create(PipelineContent.CreateContent(modelX));
```

Comment:
Same question here regarding why we need to create an instance of PipelineContent to pass to RequetsContent.Create instead of passing the model directly?